test/Main.purs

```diff
@@ -47,3 +47,18 @@ main = launchAff_ $ runSpec [ consoleReporter ] do
             , 3 /\ 3
             , 1 /\ 1
             ]
+
+  describe "build an object map via ST" do
+    it "keeps order" do
+      let m' = run (do
+            m <- STOM.new
+            _ <- STOM.poke 2 3 m
+            _ <- STOM.poke 3 3 m
+            _ <- STOM.poke 1 1 m
+            _ <- STOM.delete 3 m
```

Comment:
I think I'd remove the `delete` part from this test, so it just shows that the odd order of "2,3,1" is stable. Like in the other tests.